src/shared/fdset.c

```diff
@@ -10,23 +10,23 @@
 #include "dirent-util.h"
 #include "fd-util.h"
 #include "fdset.h"
+#include "hashmap.h"
 #include "log.h"
 #include "macro.h"
 #include "parse-util.h"
 #include "path-util.h"
-#include "set.h"
 #include "stat-util.h"
 
-#define MAKE_SET(s) ((Set*) s)
+#define MAKE_HASHMAP(s) ((Hashmap*) s)
```

Comment:
can you add a comment somewhere up top here explaining how the hashmap is organized? this deservers some explanation in the code, since it's not obvious. i.e. just say what points to what, and means what and why.